services_app/src/androidTest/java/org/opendatakit/webkitserver/service/OdkWebserverServiceTest.java

```diff
@@ -710,7 +710,8 @@ private void assertResponseForDifferentExtension(File fileLocation) {
     }
 
     private File createTestFile(File directoryLocation, String fileName, String content) {
-        File fileLocation = new File(directoryLocation, fileName);
+        Uri fileUri = Uri.withAppendedPath(Uri.fromFile(directoryLocation), fileName);
+        File fileLocation = new File(fileUri.getPath());
```

Comment:
This still uses the legacy "File" you should be using DocumentFile or the other newer Android URI classes for files.

https://www.tabnine.com/code/java/methods/androidx.documentfile.provider.DocumentFile/createFile
https://developer.android.com/reference/androidx/documentfile/provider/DocumentFile
https://stackoverflow.com/questions/59189631/creating-a-writeable-documentfile-from-uri
https://developer.android.com/training/data-storage/shared/documents-files
https://stackoverflow.com/questions/39577708/is-it-possible-to-get-a-regular-file-from-documentfile